profiler/profiler.go

```diff
@@ -315,3 +319,22 @@ type StatsdClient interface {
 	// Timing creates a distribution of the values registered as the duration of a certain event.
 	Timing(event string, duration time.Duration, tags []string, rate float64) error
 }
+
+// getAgentURLFromEnv determines the trace agent URL from environment variable
+// DD_TRACE_AGENT_URL. If the determined value is valid, it returns the value and true.
+// Otherwise, it returns an empty string and false.
+func getAgentURLFromEnv() (string, bool) {
```

Comment:
Find a way to have this function only once, not twice. Also, I think it's best to avoid the `get*` prefix: https://go.dev/doc/effective_go#Getters

Reply:
Renamed to `AgentURLFromEnv` and moved over to `internal/agent.go`. I also moved `ResolveAgentAddr` and the default consts (Hostname, Port, Address and URL).